hack/e2e.go

```diff
@@ -14,968 +14,144 @@ See the License for the specific language governing permissions and
-	deprecatedCtlCmd = flag.String("ctl", "", "Deprecated. Does nothing.")
-)
-
-const kubernetesAnywhereConfigTemplate = `
-.phase1.num_nodes=4
-.phase1.cluster_name="{{.Cluster}}"
-.phase1.cloud_provider="gce"
-
-.phase1.gce.os_image="ubuntu-1604-xenial-v20160420c"
-.phase1.gce.instance_type="n1-standard-2"
-.phase1.gce.project="{{.Project}}"
-.phase1.gce.region="us-central1"
-.phase1.gce.zone="us-central1-b"
-.phase1.gce.network="default"
-
-.phase2.installer_container="docker.io/colemickens/k8s-ignition:latest"
-.phase2.docker_registry="gcr.io/google-containers"
-.phase2.kubernetes_version="v1.4.1"
-.phase2.provider="{{.Phase2Provider}}"
-
-.phase3.run_addons=y
-.phase3.kube_proxy=y
-.phase3.dashboard=y
-.phase3.heapster=y
-.phase3.kube_dns=y
-`
-
-func appendError(errs []error, err error) []error {
-	if err != nil {
-		return append(errs, err)
-	}
-	return errs
-}
-
-func validWorkingDirectory() error {
-	cwd, err := os.Getwd()
-	if err != nil {
-		return fmt.Errorf("could not get pwd: %v", err)
-	}
-	acwd, err := filepath.Abs(cwd)
-	if err != nil {
-		return fmt.Errorf("failed to convert %s to an absolute path: %v", cwd, err)
-	}
-	// This also matches "kubernetes_skew" for upgrades.
-	if !strings.Contains(filepath.Base(acwd), "kubernetes") {
-		return fmt.Errorf("must run from kubernetes directory root: %v", acwd)
-	}
-	return nil
+type flags struct {
+	get  bool
+	old  time.Duration
+	args []string
 }
 
-type TestCase struct {
-	XMLName   xml.Name `xml:"testcase"`
-	ClassName string   `xml:"classname,attr"`
-	Name      string   `xml:"name,attr"`
-	Time      float64  `xml:"time,attr"`
-	Failure   string   `xml:"failure,omitempty"`
-}
-
-type TestSuite struct {
-	XMLName  xml.Name `xml:"testsuite"`
-	Failures int      `xml:"failures,attr"`
-	Tests    int      `xml:"tests,attr"`
-	Time     float64  `xml:"time,attr"`
-	Cases    []TestCase
-}
-
-var suite TestSuite
-
-func xmlWrap(name string, f func() error) error {
-	start := time.Now()
-	err := f()
-	duration := time.Since(start)
-	c := TestCase{
-		Name:      name,
-		ClassName: "e2e.go",
-		Time:      duration.Seconds(),
-	}
-	if err != nil {
-		c.Failure = err.Error()
-		suite.Failures++
-	}
-	suite.Cases = append(suite.Cases, c)
-	suite.Tests++
-	return err
-}
+const (
+	getDefault = true
+	oldDefault = 24 * time.Hour
+)
 
-func writeXML(start time.Time) {
-	suite.Time = time.Since(start).Seconds()
-	out, err := xml.MarshalIndent(&suite, "", "    ")
-	if err != nil {
-		log.Fatalf("Could not marshal XML: %s", err)
-	}
-	path := filepath.Join(*dump, "junit_runner.xml")
-	f, err := os.Create(path)
-	if err != nil {
-		log.Fatalf("Could not create file: %s", err)
-	}
-	defer f.Close()
-	if _, err := f.WriteString(xml.Header); err != nil {
-		log.Fatalf("Error writing XML header: %s", err)
-	}
-	if _, err := f.Write(out); err != nil {
-		log.Fatalf("Error writing XML data: %s", err)
+func parse(args []string) (flags, error) {
```

Comment:
Actually I added this unit test because it is non-trivial and I messed it up multiple times. These unit tests are what give me confidence it does the right thing.  Specifically that go run hack/e2e.go --get=false --blah --foo sends --blah and --foo to kubetest.